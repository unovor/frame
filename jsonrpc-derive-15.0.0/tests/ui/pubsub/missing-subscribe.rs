#[macro_use]
extern crate jsonrpc_derive;
extern crate jsonrpc_core;
extern crate jsonrpc_pubsub;

#[rpc]
pub trait Rpc {
	type Metadata;

	// note that a subscribe method is missing

	/// Unsubscribe from hello subscription.
	#[pubsub(subscription = "hello", unsubscribe, name = "hello_unsubscribe")]
	fn unsubscribe(&self, _: Option<Self::Metadata>, _: SubscriptionId) -> Result<bool>;
}

fn main() {}
