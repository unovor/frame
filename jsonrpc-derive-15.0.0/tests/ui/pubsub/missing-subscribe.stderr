error: subscription 'hello'. Can't find subscribe method, expected a method annotated with `subscribe` e.g. `#[pubsub(subscription = "hello", subscribe, name = "hello_subscribe")]`
  --> $DIR/missing-subscribe.rs:12:2
   |
12 |       /// Unsubscribe from hello subscription.
   |  _____^
13 | |     #[pubsub(subscription = "hello", unsubscribe, name = "hello_unsubscribe")]
14 | |     fn unsubscribe(&self, _: Option<Self::Metadata>, _: SubscriptionId) -> Result<bool>;
   | |________________________________________________________________________________________^

error: aborting due to previous error

